.github/workflows/dagster.yml

```
@@ -30,133 +56,81 @@ on:
+        name: Deploy to cluster
         run: |
-          docker run -p 8080:8080 -d --name test_container test-image \
-              dagster api grpc -h 0.0.0.0 -p 8080 --python-file "/dagster/${{ steps.find_repo_file.outputs.REPOSITORY_PY_LOCATION }}"
-          sleep 5
-          wget -O/tmp/grpc_health_probe -q https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/v0.4.11/grpc_health_probe-linux-amd64
-          chmod +x /tmp/grpc_health_probe
-          for i in $(seq 1 10); do /tmp/grpc_health_probe -addr=localhost:8080 && s=0 && break || s=$? && sleep 2; done; (docker logs test_container && exit $s)
+          kubectl rollout restart deployment -n dagster dagster-dagster-user-deployments-${{ inputs.image_name }}
+          kubectl rollout status deployment -n dagster dagster-dagster-user-deployments-${{ inputs.image_name }} -w --timeout=10m && echo "Deployment completed" || ( kubectl logs -n dagster -l deployment=${{ inputs.image_name }} && exit 1 )
 
-      - name: Push docker image
-        uses: docker/build-push-action@v4.0.0
-        env:
-          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+  deploy-dev-env-to-gke:
+    if: ${{ github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'dev-env') }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v1'
         with:
-          context: .
-          push: true
-          ssh: default
-          tags: |
-            ${{ github.event_name == 'pull_request' && format('eu.gcr.io/{0}/{1}:pr-{2}', inputs.gcp_project, inputs.image_name, github.event.number) || '' }}
-            ${{ github.ref == format('refs/heads/{0}', inputs.branch) && format('eu.gcr.io/{0}/{1}:latest', inputs.gcp_project, inputs.image_name) || '' }}
-            ${{ github.ref == format('refs/heads/{0}', inputs.branch) && format('eu.gcr.io/{0}/{1}:stage', inputs.gcp_project, inputs.image_name) || '' }}
-            ${{ startsWith(github.ref, 'refs/tags/') && steps.get_tag.outputs.IMAGE_FULLNAME || '' }}
-            ${{ startsWith(github.ref, 'refs/tags/') && format('eu.gcr.io/{0}/{1}:prod', inputs.gcp_project, inputs.image_name) || '' }}
-          provenance: false # Image attestations not supported in GCR: https://github.com/docker/buildx/issues/1533
+          credentials_json: '${{ secrets.GCR_RW_SERVICEACCOUNT_KEY }}'
 
-      - uses: google-github-actions/get-gke-credentials@v0.3.0
+      - uses: 'google-github-actions/get-gke-credentials@v1'
+        name: 'Get GKE cluster credentials'
         with:
-          cluster_name: ${{ startsWith(github.ref, 'refs/tags/') && inputs.cluster_name || format('stage-{0}', inputs.cluster_name) }}
-          location: europe-west4
+          cluster_name: ${{ startsWith(github.ref, 'refs/tags/') && inputs.cluster_name || inputs.stage_cluster_name }}
```

Could be simplified to `stage_cluster_name` since dev-env is only running in PRs
```suggestion
          cluster_name: ${{ inputs.stage_cluster_name }}
```